### ISSUE TITLE
Pride, Prejudice by @hugovk

### ISSUE THREAD

**Issue by hugovk** [NaNoGenMo/2017#130]
# *Pride, Prejudice*

## Generated output

* [output.txt](https://github.com/hugovk/NaNoGenMo-2017/blob/master/03-reducifier/output.txt): 103,038 words
* [output2.txt](https://github.com/hugovk/NaNoGenMo-2017/blob/master/03-reducifier/output2.txt): 51,142 words

## What it does

The problem isn't generating over 50,000 words. The problem is existing books are too long. *Pride and Prejudice* is 130,000 words, *Moby Dick* is 215,136 words (or [215,136 meows](https://github.com/hugovk/meow.py)). And we all know 50,000 is the gold standard for a novel! So how can we reduce the word count? 

* Remove Project Gutenberg boilerplate
* Use contractions everywhere: "won't" instead of "will not", "t'" instead of "the"
* Replace "and" with a comma, "or" with a slash
* Delete parenthetical "however", "indeed" and "I dare say"
* Remove honorifics (Mr., Mrs., Miss, Dr.)
* ['Substitute ‘damn’ every time you’re inclined to write ‘very’; your editor will delete it and the writing will be just as it should be.'](https://quoteinvestigator.com/2012/08/29/substitute-damn/)
* Replace redundant phrases like "whether or not" with just "whether"

These tactics reduce *Pride and Prejudice* by about 15% to 111,000 words. 

Next we work out the ratio of words we have to 50k, count how many sentences we have, and work out how many sentences we want to approach 50k and use a text summariser to chop out the dead wood.

## How to do it

Run:

```bash
pip install -r requirements.txt

python reducifier.py
```

Example:
```bash
python reducifier.py
open
word count: 130,000
word count: 126,936	diff: 97.643%	deboilerplatify
word count: 125,438	diff: 96.491%	remove_quote_things
word count: 121,549	diff: 93.499%	deveryify
word count: 121,018	diff: 93.091%	decontractify
word count: 111,633	diff: 85.872%	dehonorify
Ratio (words/50k):	 3
Number of sentences:	 4588
Number to keep:		 1529
word count: 54,273	diff: 41.748%	summarise
```

This produces output.txt before the summariser, and output2.txt after the summariser.

Works at least with macOS High Sierra with Python 3.6.3.

## Example

Here's a diff of *Pride and Prejudice* and the first pass output.txt:

!['tis a truth universally acknowledged, that a single man in possession of a good fortune, must be in want of a wife.](https://user-images.githubusercontent.com/1324225/33440948-0ff233c0-d5fa-11e7-9059-bc23df10ee2a.png)

**Comment by janelleshane** [comment]
Ha, this is great! 60% reduced Pride and Prejudice is still totally readable. 

Too bad the summarizer took out all the damns.

**Comment by henrikh** [comment]
"Remove honorifics (Mr., Mrs., Miss, Dr.)" 😱 How can I then tell the "Bennet"s apart?!

**Comment by alexyuriev** [comment]
@janelleshane Cliff-notes are also readable.

**Comment by sandes** [comment]
Great

**Comment by danesparza** [comment]
@henrikh Agreed -- lines [like this](https://github.com/hugovk/NaNoGenMo-2017/blob/master/03-reducifier/output.txt#L1401) become ... odd.

**Comment by hugovk** [comment]
@henrikh @danesparza Yep, I did realise that but unfortunately they just had to go to reduce the word count :)  I should have replaced "Mrs. Bennet" with her maiden name, "Gardiner"!

**Comment by bryanrasmussen** [comment]
Sometimes you will see major characters referred to with a shortened version of the name after introduction. I would suggest calling Mrs. Bennet Mrs. B, Mr. Bennet Mr. B. You don't remove honorifics and reduce word count but you reduce character count.

**Comment by bryanrasmussen** [comment]
Actually considering the patriarchy Mr. B can just be B.

**Comment by hugovk** [comment]
@bryanrasmussen Word count is all that matters :)

**Comment by bryanrasmussen** [comment]
Not if your last name is Hugo, and your first Victor!

**Comment by hugovk** [comment]
> > PS. Using the 't' contraction instead of 'the' makes this really hard to parse.
>
> Only in some cases.
>
> "...by a young man of large fortune from t'north of England;"[1]
>
> This is just about the perfect edit.
>
> [1] https://github.com/hugovk/NaNoGenMo-2017/blob/master/03-reducifier/output.txt#L35

:)

See https://news.ycombinator.com/item?id=15823499 for more discussion.

**Comment by henrikh** [comment]
@philsnow As far as I recall, Elizabeth is actually referred to as "Miss Bennet" when adressed _directly_ by Mr Darcy and Mr Wickham -- but, of course, in those situations there would be no doubt :wink:
